Add actionBuildTemplateVariables hook

PrestaEdit opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project!

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/#pull-requests

For type and category see:
https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/pull-requests/#type--category
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Allow to add new preview variables while preview a mail.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | To be tested by a developer, check below
| Related PRs       | 

## How to test?

Add a module with this hook :
```
public function hookActionBuildTemplateVariables($params)
    {
        return [
            '{name}' => 'test'
        ];
    }
```

Then go to backoffice > Design > Email theme ; go at the bottom and click one glass to reach backoffice page "Previewing theme (...)". There you can click on emails 'HTTP' icon to preview an email and trigger the hook. Please verify available smarty variables in Smarty

PrestaEdit committed:
Add actionBuildTemplateVariables hook

PrestaEdit committed:
Add missing use

matks commented on src/Adapter/MailTemplate/MailPreviewVariablesBuilder.php:
```diff
@@ -146,6 +148,20 @@ public function buildTemplateVariables(LayoutInterface $mailLayout)
         $templateVars['{color}'] = $this->configuration->get('PS_MAIL_COLOR');
         $templateVars = array_merge($templateVars, $this->buildOrderVariables($mailLayout));
 
+        if ($mailLayout->getModuleName() !== '') {
```
I am not sure to understand why this check

PrestaEdit replied:
It's to allow to add only related vars to the specific layout coming from a module ; it's not needed to add specific layout var - as I think - for another layours than the module itself.

Hlavtox replied:
What if the name is false or null? It would be better to add empty.

matks reviewed (commented):
The code seems globally OK but it relies on the fact `$mailLayout->getModuleName()` is not an empty string and I don't know how that happens, can you please provide context?

I also improved the "How to test"

Hlavtox reviewed (changes requested):
Please, add MailPreview to the hook name. This can be confused with some hooks to add vars to FO.

PrestaEdit commented:
It's an **action** hook, not a **display** one.

PrestaEdit committed:
Update hook name for mail preview template variables

Changed the hook from 'actionBuildTemplateVariables' to 'actionMailPreviewBuildTemplateVariables' in MailPreviewVariablesBuilder to better reflect its purpose and avoid conflicts with other template variable hooks.

PrestaEdit commented on src/Adapter/MailTemplate/MailPreviewVariablesBuilder.php:
```diff
@@ -146,6 +148,20 @@ public function buildTemplateVariables(LayoutInterface $mailLayout)
         $templateVars['{color}'] = $this->configuration->get('PS_MAIL_COLOR');
         $templateVars = array_merge($templateVars, $this->buildOrderVariables($mailLayout));
 
+        if ($mailLayout->getModuleName() !== '') {
```
```suggestion
        if (!empty($mailLayout->getModuleName())) {
```

PrestaEdit committed:
Update src/Adapter/MailTemplate/MailPreviewVariablesBuilder.php

PrestaEdit commented:
Description and title updated:

- Title now reflects the actual hook name shipped in the diff (`actionMailPreviewBuildTemplateVariables`) — the original wording pre-dated @Hlavtox's rename request.
- **How to test** rewritten with a full drop-in test module (module file + `hello.html` + `hello.txt`) and 5 explicit BO steps. @AureRita — this should reproduce cleanly this time; the previous test failed because `ps_quality`'s own templates weren't the preview target, and the hook is intentionally module-scoped.
- Regarding @Hlavtox's earlier suggestion to also fire the hook for **core** templates: kept as a follow-up. Broadcasting a hook without `$module_id` scoping on every preview has different semantics and deserves its own PR. Happy to open it once this one lands.

@nicosomb your approval is still on the previous revision — a quick re-look would be appreciated.
